snyk/snyk
# refactor: pull out custom rules code

#### What does this PR do?

This PR extracts the custom and internal Snyk rules code from the main `test` function into its own file. For readability and in preparation of adding extended Terraform language support, which will make this function even bigger.

#### How should this be manually tested?
1. `npm run build`
2. Create a custom rules bundle using https://github.com/snyk/snyk-iac-rules: `snyk-iac-rules build ./fixtures/custom-rules -o bundle.tar.gz`
3. Run the CLI on a file that would trigger the rule (just use the fixtures in that repo): `snyk-dev iac test ./fixtures/custom-rules/rules/CUSTOM-1/fixtures`

The acceptance tests would verify this plus the remote bundles, so if the tests pass then this is good to go.
